framework/src/main/java/org/apache/drill/test/framework/TestVerifier.java

```diff
@@ -340,13 +352,29 @@ private String printSummary(List<ColumnList> unexpectedList,
     return sb.toString();
   }
 
-  private static int getMissingCount(Map<ColumnList, Integer> map) {
+  private static int getMissingCount(Map<ColumnList, Integer> map, Map<ColumnList, Integer> actualMap) {
     int missingCount = 0;
-    Iterator<Integer> iterator = map.values().iterator();
+    Iterator<Map.Entry<ColumnList, Integer>> iterator = map.entrySet().iterator();
+    
     while (iterator.hasNext()) {
-      missingCount += iterator.next();
+      Map.Entry<ColumnList, Integer> entry = iterator.next();
+      ColumnList cl = entry.getKey();
+      int count = entry.getValue();
+      if(actualMap.containsKey(cl)){
+        int actualCount = actualMap.get(cl);
+	if(count > actualCount){
+	  missingCount += (count - actualCount);
+	  entry.setValue((count - actualCount));
```

Comment:
not clear on how you are setting the expectedMap. This isn't used anywhere and not visible elsewhere, afaik.